### PR TITLE
pkg/endpoint: Consistently use logrus fields

### PR DESCRIPTION
**Pull request by raybejjani**
This switches a bunch of log callsites to use logrus.WIthFields. These changes are not supposed to break anything and follow the same reasoning from https://github.com/cilium/cilium/pull/1801 The goal of these changes is to make debugging via logs easier by making our usage more consistent. Ideally, field names should be used the same way throughout the code and mean the same thing.

The most important things to review are whether the field names make sense in the context they're used, and whether we should introduce new ones to pkg/logfields/logfields.go. I'm also happy to incorporate better messages into this PR :)
I might have been too aggressive by making a logger method bound to endpoint. I'm not sure if the fields reported are always appropriate, but they seemed to be. In particular, I'm not sure if Iteration is relevant, and if I'm reporting it correctly in any case.

### CODE REVIEW

**Review by aanm — APPROVED** [review]
Just make sure the `log()` method is always called with the  endpoint's RLock locked

**Comment by aanm** [comment]
I'm adding a WIP, once you make sure the log is being called inside the RLock then you can remove the label

File: pkg/endpoint/log.go
Comment (aanm) [inline pkg/endpoint/log.go]: @tgraf should be `e.Consumable.Iteration` or `e.policyRevision`?

Comment (tgraf) [reply pkg/endpoint/log.go]: The iteration is unused now, it predated policy revision. We can remove the iteration field and use the policy revision in the logs.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by raybejjani** [comment]
I'm still trying to figure out how to do this well. Unfortunately, we seem to lock the Endpoint somewhat arbitrarily and I need to safely access the struct. In most cases, it is Locked or RLocked, but in the Endpoint.SetIdentity case it is in some callsites and not in one: Daemon.SyncState (which calls syncLabels which calls Endpoint.SetIdentity, https://github.com/cilium/cilium/blob/master/daemon/state.go#L70-L77).

I'll wait on CI to see if I introduced an obvious deadlock, but I need to check all callees to make sure. I'll be back ;)

**Comment by raybejjani** [comment]
Ok, this seemed to pass tests. I don't think I've introduced a new deadlock.
@tgraf @aanm Could you take another look, this PR is now bit more complex.
I'll squash this to three commits before merging:
daemon: Daemon.SyncState locks Endpoint to syncLabels
pkg/endpoint: Unexport Endpoint critical path functions
pkg/endpoint: Consistently use logrus fields

**Review by tgraf — APPROVED** [review]

**Pull request merged**
